### PR TITLE
Make sure bad owasm code cannot ruin the chain with bad size

### LINKED ISSUE #583: Make sure bad owasm code cannot ruin the chain with bad "size" variables
Whether it's external data size, calldata size, return data size, raw request size, etc. Please write good tests too.

### PR DESCRIPTION
**Pull request by taobun**
fixed #583

### CODE REVIEW

File: chain/owasm/types.go [hunk 1]
```diff
@@ -33,6 +33,14 @@ type ExecutionEnvironment interface {
 	// at the specified index.
 	GetValidatorAddress(validatorIndex int64) ([]byte, error)
 
+	// GetMaximumResultSize returns a maxixmum size of result data that returns from
```
Comment (sorawit) [inline chain/owasm/types.go]: the maximum

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — DISMISSED** [review]
Looks great.

File: chain/owasm/execute_test.go [hunk 2]
```diff
@@ -82,8 +94,54 @@ func TestAllocateFailWithExceedMemory(t *testing.T) {
...
+func TestExecuteInvalidGetMaximumResultSize(t *testing.T) {
+	code, err := ioutil.ReadFile("./res/silly.wasm")
+	require.Nil(t, err)
+	env := &mockExecutionEnvironment{
+		externalDataResults:               [][][]byte{nil, {[]byte("RETURN_DATA")}},
+		requestExternalDataResultsCounter: [][]int64{nil, []int64{0}},
+		maximumResultSize:                 10,
+	}
+
+	// It should return error because "band-protocol" has length equal to 13.
```
Comment (evilpeach) [inline chain/owasm/execute_test.go]: Is it the correct comment?

Comment (taobun) [reply chain/owasm/execute_test.go]: What 's wrong?

Comment (evilpeach) [reply chain/owasm/execute_test.go]: `// It should return error because "RETURN_DATA" has length equal to 11.`

**Review by sorawit — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**